pika/compat/__init__.py

```diff
@@ -167,51 +167,35 @@ def get_linux_version(release_str):
-    # Origin: https://gist.github.com/4325783, by Geert Jansen.  Public domain.
-    def socketpair(family=socket.AF_INET, type=socket.SOCK_STREAM, proto=0):
-        if family == socket.AF_INET:
-            host = _LOCALHOST
-        elif family == socket.AF_INET6:
-            host = _LOCALHOST_V6
-        else:
-            raise ValueError(
-                'Only AF_INET and AF_INET6 socket address families '
-                'are supported')
-        if type != socket.SOCK_STREAM:
-            raise ValueError('Only SOCK_STREAM socket type is supported')
-        if proto != 0:
-            raise ValueError('Only protocol zero is supported')
-
-        # We create a connected TCP socket. Note the trick with
-        # setblocking(False) that prevents us from having to create a thread.
-        lsock = socket.socket(family, type, proto)
+def socketpair(family=socket.AF_INET, type=socket.SOCK_STREAM, proto=0):
+    if family == socket.AF_INET:
+        host = _LOCALHOST
+    elif family == socket.AF_INET6:
+        host = _LOCALHOST_V6
+    else:
+        raise ValueError(
+            'Only AF_INET and AF_INET6 socket address families '
+            'are supported')
+    if type != socket.SOCK_STREAM:
+        raise ValueError('Only SOCK_STREAM socket type is supported')
+    if proto != 0:
+        raise ValueError('Only protocol zero is supported')
+
+    lsock = socket.socket(family, type, proto)
+    try:
+        lsock.bind((host, 0))
+        lsock.listen(min(socket.SOMAXCONN, 128))
+        # On IPv6, ignore flow_info and scope_id
+        addr, port = lsock.getsockname()[:2]
+        csock = socket.socket(family, type, proto)
         try:
-            lsock.bind((host, 0))
-            lsock.listen(min(socket.SOMAXCONN, 128))
-            # On IPv6, ignore flow_info and scope_id
-            addr, port = lsock.getsockname()[:2]
-            csock = socket.socket(family, type, proto)
-            try:
-                csock.setblocking(False)
-                if _sys.version_info >= (3, 0):
-                    try:
-                        csock.connect((addr, port))
-                    except (BlockingIOError, InterruptedError):
-                        pass
-                else:
-                    try:
-                        csock.connect((addr, port))
-                    except SOCKET_ERROR as e:
-                        if e.errno != errno.EWOULDBLOCK:
-                            raise
-                csock.setblocking(True)
-                ssock, _ = lsock.accept()
-            except Exception:
-                csock.close()
-                raise
-        finally:
-            lsock.close()
-        return (ssock, csock)
-
-    socket.socketpair = socketpair
+            csock.connect((addr, port))
+            ssock, _ = lsock.accept()
+        except Exception:
+            csock.close()
+            raise
+    finally:
+        lsock.close()
+    csock.setblocking(False)
+    ssock.setblocking(False)
```

Comment:
My 2 cents: this function's name is exactly like the real deal in `socket` module which returns blocking sockets; if some other part of pika or pika's user use it, they might expect that it behaves just like the one in `socket`. For this reason, I would add the logic that makes the sockets non-blocking in the specialized method `_get_interrupt_pair` augmented with a comment, such as "make sockets non-blocking to avoid potential deadlock".